erigon-lib/diagnostics/stages.go

```diff
@@ -124,8 +130,6 @@ func (d *DiagnosticClient) runCurrentSyncStageListener(rootCtx context.Context)
 				d.mu.Lock()
```

Comment:
Also those kind of mutexes has to be stored right before the field they are protecting. Exported methods (starting from Capital letter) expected to be called somewhere outside so caller should bother with mutexes. Instead you can create an unexported method to make thread unsafe update and another exported method which wraps unsafe call with mutexes. This works good when you need to use both thread-safe and thread-unsafe methods.

Reply:
fixed